src/adapter/src/catalog/builtin_table_updates.rs

```diff
@@ -745,4 +745,42 @@ impl CatalogState {
             diff,
         }
     }
+
+    pub fn pack_storage_usage_update(
+        &self,
+        event_details: &EventDetails,
```

Comment:
This should take a VersionedEvent just like `pack_audit_log_update` so it can get the timestamp right from it. But unlike `pack_audit_log_update`, this function should unpack and assert the inner event type is the expected storage update, otherwise panic.